fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/JoinedRecordTypeTest.java

```diff
@@ -0,0 +1,208 @@
+            typeBuilder.addConstituent(OTHER_RECORD, "MyOtherRecord");
+            typeBuilder.addJoin(SIMPLE_RECORD, field("other_rec_no"), OTHER_RECORD, field("rec_no"));
+        };
+    }
+
+    @Test
+    void loadSyntheticRecord() throws ExecutionException, InterruptedException {
+        final Tuple joinedPrimaryKey;
+        final FDBStoredRecord<Message> simpleRecord;
+        final FDBStoredRecord<Message> otherRecord;
+        try (FDBRecordContext context = openContext()) {
+            createOrOpenRecordStore(context, baseMetaData(addJoinedType()));
+
+            Tuple syntheticRecordTypeKey = recordStore.getRecordMetaData()
+                    .getSyntheticRecordType(JOINED_RECORD_NAME)
+                    .getRecordTypeKeyTuple();
+            simpleRecord = recordStore.saveRecord(createSimpleRecord(100, 10));
+            otherRecord = recordStore.saveRecord(createOtherRecord(101, 11));
+            joinedPrimaryKey = Tuple.from(
+                    syntheticRecordTypeKey.getItems().get(0),
+                    simpleRecord.getPrimaryKey().getItems(),
+                    otherRecord.getPrimaryKey().getItems());
+
+            context.commit();
+        }
+
+        try (FDBRecordContext context = openContext()) {
+            createOrOpenRecordStore(context, baseMetaData(addJoinedType()));
+
+            final FDBSyntheticRecord rec = recordStore.loadSyntheticRecord(joinedPrimaryKey).get();
+            assertEquals(2, rec.getConstituents().size());
+
+            assertEquals(simpleRecord, rec.getConstituent(SIMPLE_RECORD));
+            assertEquals(otherRecord, rec.getConstituent(OTHER_RECORD));
+
+            context.commit();
+        }
+    }
+
+    @Test
+    void loadSyntheticRecordFailsMissingConstituent() throws ExecutionException, InterruptedException {
+        final Tuple joinedPrimaryKey;
+        final FDBStoredRecord<Message> simpleRecord;
+        final FDBStoredRecord<Message> otherRecord;
+        try (FDBRecordContext context = openContext()) {
+            createOrOpenRecordStore(context, baseMetaData(addJoinedType()));
+
+            Tuple syntheticRecordTypeKey = recordStore.getRecordMetaData()
+                    .getSyntheticRecordType(JOINED_RECORD_NAME)
+                    .getRecordTypeKeyTuple();
+            simpleRecord = recordStore.saveRecord(createSimpleRecord(100, 10));
+            otherRecord = recordStore.saveRecord(createOtherRecord(101, 11));
+            joinedPrimaryKey = Tuple.from(
+                    syntheticRecordTypeKey.getItems().get(0),
+                    simpleRecord.getPrimaryKey().getItems(),
+                    otherRecord.getPrimaryKey().getItems());
+
+            context.commit();
+        }
+
+        try (FDBRecordContext context = openContext()) {
+            createOrOpenRecordStore(context, baseMetaData(addJoinedType()));
+
+            recordStore.deleteRecord(otherRecord.getPrimaryKey());
```

Comment:
Is it worth testing again with `simpleRecord` deleted?

Reply:
Sure, done.